lib/docker/imageservice.go

```diff
@@ -336,29 +382,59 @@ func initTransport(req RegistryConnectionRequest) (*http.Transport, string) {
 		DisableKeepAlives:   true,
 	}
 
-	cert, err := tls.LoadX509KeyPair(req.ClientCertPath, req.ClientKeyPath)
-	if err == nil {
-		roots, err := newCertPool([]string{req.CACertPath})
-		if err == nil {
+	// Figure out the registry address scheme (http or https). If the scheme
+	// was specified explicitly, keep it as-is. Otherwise, default to https
+	// unless the insecure flag was provided.
+	defaultScheme := "https"
+	if req.Insecure {
```

Comment:
Maybe it's a docker thing, but usually Insecure implies turning off TLS certificate chain validation, not changing the scheme to http.

Reply:
Yeah, in Docker "insecure" actually kinda means both things:
* Try to use https and ignore the certificate error.
* If https is not available, use http.
I guess we could default to https and require "http://" explicitly for "insecure-insecure" registries.